packages/node/test/integration/cant-take-action.spec.ts

```diff
@@ -0,0 +1,195 @@
+import {
+  Address,
+  AppABIEncodings,
+  AppState,
+  AssetType,
+  Node as NodeTypes
+} from "@counterfactual/types";
+import { Contract, ContractFactory, Wallet } from "ethers";
+import { AddressZero, One, Zero } from "ethers/constants";
+import { BaseProvider, Web3Provider } from "ethers/providers";
+import { hexlify, randomBytes, SigningKey } from "ethers/utils";
+import FirebaseServer from "firebase-server";
+import ganache from "ganache-core";
+import { v4 as generateUUID } from "uuid";
+
+import TicTacToeApp from "../../../apps/build/TicTacToeApp.json";
+import {
+  IMessagingService,
+  InstallMessage,
+  IStoreService,
+  Node,
+  NODE_EVENTS,
+  NodeConfig,
+  ProposeMessage
+} from "../../src";
+import { ERRORS } from "../../src/methods/errors";
+
+import TestFirebaseServiceFactory from "./services/firebase-service";
+import {
+  EMPTY_NETWORK,
+  generateTakeActionRequest,
+  getNewMultisig,
+  makeInstallRequest
+} from "./utils";
+
+describe("Node method follows spec - takeAction", () => {
+  let firebaseServiceFactory: TestFirebaseServiceFactory;
+  let firebaseServer: FirebaseServer;
+  let messagingService: IMessagingService;
+  let nodeA: Node;
+  let storeServiceA: IStoreService;
+  let nodeB: Node;
+  let storeServiceB: IStoreService;
+  let nodeConfig: NodeConfig;
+  let provider: BaseProvider;
+  let tttContract: Contract;
+  let privateKey: string;
+
+  beforeAll(async () => {
+    firebaseServiceFactory = new TestFirebaseServiceFactory(
+      process.env.FIREBASE_DEV_SERVER_HOST!,
+      process.env.FIREBASE_DEV_SERVER_PORT!
+    );
+    firebaseServer = firebaseServiceFactory.createServer();
+    messagingService = firebaseServiceFactory.createMessagingService(
+      process.env.FIREBASE_MESSAGING_SERVER_KEY!
+    );
+    nodeConfig = {
+      STORE_KEY_PREFIX: process.env.FIREBASE_STORE_PREFIX_KEY!
```

Comment:
Minor comment but it might be cleaner stylistically to have a single file called `env.spec.ts` or something where we fetch all these variables once and them import them symbolically as `TestEnvironment`

Reply:
Yeah, that's cleaner - I'll put up another PR for that.